processor/src/test/java/org/mapstruct/ap/test/collection/iterabletononiterable/IterableToNonIterableMappingTest.java

```diff
@@ -45,4 +47,34 @@ public void shouldReverseMapStringListToStringUsingCustomMapper() {
         assertThat( source.getNames() ).isEqualTo( Arrays.asList( "Alice", "Bob", "Jim" ) );
         assertThat( source.publicNames ).isEqualTo( Arrays.asList( "Alice", "Bob", "Jim" ) );
     }
+
+    @Test
+    @IssueKey("607")
+    @WithClasses({
+            FruitsMapper.class
+    })
+    public void shouldMapIterableToNonIterable() {
+        List<Fruit> fruits =  Arrays.asList( new Fruit( "mango" ), new Fruit( "apple" ),
+                new Fruit( "banana" ) );
+        FruitsMenu menu = new FruitsMenu(fruits);
+        FruitSalad salad = FruitsMapper.INSTANCE.fruitsMenuToSalad( menu );
+        assertThat( salad.getFruits().get( 0 ).getType() ).isEqualTo( "mango" );
+        assertThat( salad.getFruits().get( 1 ).getType() ).isEqualTo( "apple" );
+        assertThat( salad.getFruits().get( 2 ).getType() ).isEqualTo( "banana" );
```

Comment:
You can do it in one assertion by using `assertThat( salad.getFruits() ).extracting( Fruit::getType ).containsExactly( "mango", "apple", "banana" )`. Same comment stands for the test method below.